### PR TITLE
[PW-5815] Handle declined flow

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**

With this pr we are adding the `handleDeclineFlow` which redirects from checkout page back to the amazon pay page to choose another card. For this case the response includes the declined payment information. 
Testing using the amazon pay declined cards returns only an error message `AmazonPay internal error: Unknown issue has occurred while processing. `  For this case the component is remounted with the current return url using the url `amazonsessionkey` 

**Tested scenarios**

- [x] Amazon pay Happy flow 
- [x] Choose a declined card
- [x] Decline from Adyen with test code 
- [x] 3DS Declined - wrong password
- [x] 3DS Happy flow Amazon pay
- [x] From Declined to happy flow
- [x] CC 3DS2
- [x] CC

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js [hunk 1]
```diff
@@ -750,7 +747,18 @@ define(
                 if (paymentMethod.methodIdentifier.includes('amazonpay')) {
                     configuration.productType = 'PayAndShip';
                     configuration.checkoutMode = 'ProcessOrder';
-                    configuration.returnUrl = location.href;
+                    var url = new URL(location.href);
+                    if (url.searchParams.has('amazonCheckoutSessionId')) {
+                        url.searchParams.delete('amazonCheckoutSessionId');
+                    }
```
Comment (michaelpaul) [inline view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js:753]: We can omit the `has` check and always `delete`. If the parameter does not exist, `delete` accomplished it's job the easy way. :) 
```suggestion
                    url.searchParams.delete('amazonCheckoutSessionId');
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js [hunk 2]
```diff
@@ -460,7 +457,7 @@ define(
                     mount(actionNode);
                 }
             },
-            handleOnSubmit: function(state, component) {
+            handleOnSubmit: async function(state, component) {
```
Comment (michaelpaul) [inline view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js:460]: Is this method different from the one we use at line `125`? If it's the same, we might need an `await` there too.

File: view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js [hunk 3]
```diff
@@ -789,25 +797,24 @@ define(
                                 currency: configuration.amount.currency,
                                 value: configuration.amount.value
                             },
-                            returnUrl: location.href,
                             showChangePaymentDetailsButton: false
                         }).mount(containerId);
                         amazonPayComponent.submit();
                         result.component = amazonPayComponent;
-                    } else {
```
[the remark this answers is not in the document]
Comment (AlexandrosMor) [reply view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js:797]: Yes the `else` was removed intentional. In any time it should create and mount the checkout component. The issue was that after amazon pay the component was stuck with the sessionId and it was not recreated so the button was missing for declined flow

**Review by michaelpaul — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Comment by AlexandrosMor** [comment]
Thank you for reviewing. We scheduled a meeting with amazon and after testing with them I am going to merge this

**Pull request merged**